src/Compression/CompressionCodecGorilla.cpp

```diff
@@ -280,6 +281,9 @@ void decompressDataForType(const char * source, UInt32 source_size, char * dest)
     if (source + sizeof(T) > source_end || items_count < 1)
         return;
 
+    if (dest + items_count * sizeof(T) > dest_end)
```

Comment:
Is overflow possible here?

Reply:
We can read any value into `items_count` from the data sent by the user, so the overflow seems to be easily possible.

Reply:
Ok, let's check for it with `arithmeticOverflow` or `__builtin_mul_overflow`

Reply:
For example in the added test the `\xff\xff\xff\xff\` part is the value that would be read into `items_count`

Reply:
Oh, the regular overflow